crates/nu-cli/src/completions/completer.rs

```diff
@@ -545,23 +505,25 @@ impl NuCompleter {
                             let ctx = Context::new(working_set, span, b"", offset);
                             let results = self.process_completion(&mut completion, &ctx);
 
-                            // Prioritize flag completions above everything else
-                            let flags_length = flag_completions.len();
-                            suggestions.splice(0..0, flag_completions);
-
                             // Prioritize external results over (sub)commands
-                            suggestions.splice(flags_length..flags_length, results);
+                            suggestions.splice(0..0, results);
 
-                            if !completion.need_fallback {
+                            let arg_prefix = working_set.get_span_contents(span);
+                            // Still need internal flag name completions,
+                            // but not flag value completions
+                            if !(completion.need_fallback
+                                || (arg_prefix.starts_with(b"-")
+                                    && !matches!(arg, Argument::Named((_, _, Some(val))) if val.span.contains(pos))))
```

Comment:
nitpick: This condition's making my head spin. The comment above does explain it, but can we simplify it a bit, maybe something like this?
```
let is_flag_name =
    arg_prefix.starts_with(b"-") &&
        !matches!(arg, Argument::Named((_, _, Some(val))) if val.span.contains(pos));
if !completion.need_fallback && !is_flag_name
```